src/submodule.c

```diff
@@ -416,7 +458,7 @@ typedef struct {
 	git_repository *repo;
 } lfc_data;
 
-static int all_submodules(git_repository *repo, git_strmap *map)
+int all_submodules(git_repository *repo, git_strmap *map)
```

Comment:
Now that this is public (within the library anyway), this should have a name that indicates it's internal to the library.  Something like `git_submodule__all` or `git_submodule__map`, whichever makes more sense to you.